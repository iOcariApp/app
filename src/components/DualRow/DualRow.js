import React from "react";
import { View } from "react-native";
import PropTypes from "prop-types";
import styles from "./DualRow.style";

const DualRow = ({ left, right, separation, style }) => (
  <View style={[styles.main, style]}>
    <View style={[styles.column, { marginRight: separation }]}>{left}</View>
    <View style={styles.column}>{right}</View>
  </View>
);

DualRow.propTypes = {
  left: PropTypes.node,
  right: PropTypes.node,
  separation: PropTypes.number,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
};
DualRow.defaultProps = {
  separation: 0,
  style: {},
};

export default DualRow;
