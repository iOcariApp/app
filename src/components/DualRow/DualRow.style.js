import { StyleSheet } from "react-native";

export default StyleSheet.create({
  main: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "flex-start",
    width: "100%",
  },
  column: {
    flex: 1,
  },
});
