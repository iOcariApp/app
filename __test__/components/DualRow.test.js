import React from "react";
import { Text } from "react-native";
import renderer from "react-test-renderer";

import DualRow from "components/DualRow";

test("renders correctly", () => {
  const tree = renderer
    .create(
      <DualRow
        left={<Text>left</Text>}
        right={<Text>right</Text>}
        separation={10}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("applies `style` prop", () => {
  const tree = renderer
    .create(
      <DualRow
        left={<Text>left</Text>}
        right={<Text>right</Text>}
        separation={10}
        style={{ width: "50%" }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
