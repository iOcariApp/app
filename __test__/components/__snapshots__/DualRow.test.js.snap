// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies \`style\` prop 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
      Object {
        "width": "50%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "marginRight": 10,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      left
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      right
    </Text>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "marginRight": 10,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      left
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      right
    </Text>
  </View>
</View>
`;
